### DISCUSSION TITLE
Support for language localisation - Error / Exception messages

### DISCUSSION THREAD

**Discussion by vicenterusso** [spatie/laravel-permission#1686] [Ideas]
It would be nice to have a way to translate error messages from "UnauthorizedException" for example

**Comment by drbyte** [comment]
@Gummibeer have you any insights on this with other packages you're working on?

**Reply by Gummibeer** [reply]
We don't do this anywhere.
I also don't see the reason as exception messages are meant for the developer and not the public.
The messages themselves are also not really part of the public API which they would get as soon as they are translated.

In general it's up to the user to display pretty messages to his users and not the package.

If it isn't done already it would be a possibility to use custom exceptions like `MissingPermissionException` which extends the `UnauthorizedException` and contain the related data like permission name and user model instance.
So the user has all data to generate a pretty message.

Okay, I've checked the custom exceptions and it seems like they provide everything needed to the user.

**Reply by drbyte** [reply]
Thanks. I feel the same way.
Good suggestion on the re-wrapping of the exception in userland if truly requiring modification.
